src/main/java/com/palantir/gradle/versions/MavenRepositories.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions;
+
+import com.fasterxml.jackson.annotation.JsonRootName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import java.util.List;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonDeserialize(as = ImmutableMavenRepositories.class)
+@JsonSerialize(as = ImmutableMavenRepositories.class)
+@JsonRootName("repositories")
+interface MavenRepositories {
```

Comment:
I expect we're going to want to use this code to parse the XML in the gradle-consistent-versions-idea-plugin repo... unless we end up parsing it using some intellij mechanism. If so, it would make sense to put this in it's own project so the idea-plugin can include just this classes/dependencies

Reply:
It might be worth looking at how palantir-java-format does this actually (look at `.idea/palantir-java-format.xml`). It makes a plugin settings page for itself and we configure that file. It would expose the repos in the UI, which idk is good or bad.

Reply:
So after looking into this I have switched to creating / updating a idea plugin settings xml that can now be easily accessed and modified from within plugin - this has let me add a project level disable button for projects that require it see https://github.com/palantir/gradle-consistent-versions-idea-plugin/pull/27